src/common/io.py

```diff
@@ -1344,16 +1381,30 @@ def _get_trajectory(self, data_index):
 
     def _get_diagnostics_trajectory(self, data_index):
         """ Returns trajectory for the diagnostics variable that corresponds to index 'data_index'. """
+        if self._mtime != os.path.getmtime(self._fname):
+            if self._allow_file_updates:
+                self._update_data_info()
+            else:
+                raise JIOError("The result file have been modified since the result object was created. Please make sure that different filenames are used for different simulations.")
```

Comment:
Same here regarding the duplicated code as my previous comment.

Reply:
Fixed!